src/util/prolog/prologSession.ts

```diff
@@ -23,7 +24,7 @@ function resolveImports(program: string, defaultPath: string, currentFileName: s
 		let importedPrograms: string = "";
 		importList.forEach((importFileName: string) => {
 			const currentImport = fs.readFileSync(path.resolve(defaultPath, importFileName + ".pl"), "utf-8");
-			importedPrograms = importedPrograms + currentImport;
+			importedPrograms = importedPrograms + PSM.fixVariableShadowingInImport(importedPrograms + program, currentImport);
```

Comment:
Just curious: How does the "+"-operator between importedPrograms and program work? Just a concatenation?

Reply:
I think string-"addition" is common enough to be understood.